lib/os/heap-validate.c

```diff
@@ -386,3 +386,60 @@ void sys_heap_print_info(struct sys_heap *heap, bool dump_chunks)
 {
 	heap_print_info(heap->heap, dump_chunks);
 }
+
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+int heap_runtime_stats_get(struct sys_heap *heap,
+		struct sys_heap_runtime_stats *stats)
+{
+	size_t free_bytes = 0;
+	struct z_heap *h;
+	chunkid_t c = 0;
+
+	if ((heap == NULL) || (stats == NULL)) {
+		return -EINVAL;
+	}
+
+	h = heap->heap;
+	stats->free_bytes = 0;
+	stats->allocated_bytes = 0;
+	stats->max_avail_bytes = 0;
+
+	/* Iterate through all chunks */
+	do {
+		if (chunk_used(h, c)) {
+			if ((c != 0) && (c != h->end_chunk)) {
+				stats->allocated_bytes += chunksz_to_bytes(
+						h, chunk_size(h, c));
```

Comment:
Can we put the two lines in one line?

Reply:
According to the Zephyr Coding Guidelines all if-else if constructs shall be terminated with an else statement.

Reply:
thanks, I have added some empty else {; } for coding guideline.

Reply:
yes, we can put them into one line like below,
```
if (chunk_used(h, c) && (c != 0) && (c != h->end_chunk)) {
...
} else if (!chunk_used(h, c) && !solo_free_header(h, c))
...
}
```
but I think using chunk_used(h, c) to distinguish free and allocated firstly is more obvious.

Reply:
I just meant lines 412 and 413.

Reply:
that's because the max length can't be more than 80 chars.